src/OpenSSL/crypto.py

```diff
@@ -328,7 +340,7 @@ def from_cryptography_key(cls, crypto_key: _Key) -> "PKey":
             PublicFormat,
         )
 
-        if isinstance(crypto_key, (rsa.RSAPublicKey, dsa.DSAPublicKey)):
+        if hasattr(crypto_key, "public_bytes"):
```

Comment:
I'd prefer we list our all of the public key types here. There's no guarantee that we'll never add a public bytes method to a private key.

Reply:
Fine by me, I'll make the requested change.